pyomo/gdp/plugins/binary_multiplication.py

```diff
@@ -0,0 +1,175 @@
+from .gdp_to_mip_transformation import GDP_to_MIP_Transformation
+from pyomo.common.config import ConfigDict, ConfigValue
+from pyomo.core.base import TransformationFactory
+from pyomo.core.util import target_list
+from pyomo.gdp import Disjunction
+from pyomo.gdp.plugins.gdp_to_mip_transformation import GDP_to_MIP_Transformation
+from pyomo.core.util import target_list
+from weakref import ref as weakref_ref
+import logging
+
+
+logger = logging.getLogger('pyomo.gdp.binary_multiplication')
+
+
+@TransformationFactory.register(
+    'gdp.binary_multiplication',
+    doc="Reformulate the GDP as an MINLP by multiplying f(x) <= 0 by y to get f(x) * y <= 0.",
+)
+class GDPToMINLPTransformation(GDP_to_MIP_Transformation):
+    CONFIG = ConfigDict("gdp.binary_multiplication")
+    CONFIG.declare(
+        'targets',
+        ConfigValue(
+            default=None,
+            domain=target_list,
+            description="target or list of targets that will be relaxed",
+            doc="""
+
+        This specifies the list of components to relax. If None (default), the
```

Comment:
This is my typo propagating, probably, but 'transform' would be a clearer verb than 'relax'